Bootstrap scripts fail on second run due to incorrect OU ID retrieval

## Description

When running bootstrap scripts for the second time (after initial setup), the `01-default-resources.sh` script fails with an "Organization unit mismatch" error during admin user creation.

## Error Message

```
[ERROR] ✗ Failed to create admin user (HTTP 400)
Response: {"code":"USR-1023","message":"Organization unit mismatch","description":"The organization unit does not match the user type configuration"}
```

## Root Cause

The issue occurs in the OU conflict handling logic:

1. **First run**: Creates "Default" OU and "Person" user schema with matching OU IDs → Works fine
2. **Second run**:
   - Script tries to create "Default" OU → Gets HTTP 409 (already exists)
   - Script calls `GET /organization-units` to retrieve all OUs
   - **Bug**: Extracts the **first OU ID** from response using `grep -o '"id":"[^"]*"' | head -1`
   - If multiple OUs exist (e.g., "Customers" from other scripts), the first ID might not be the "Default" OU
   - Admin user creation uses wrong OU ID, but "Person" schema still references the original OU ID
   - Validation fails: `organizationUnitID != schemaOrganizationUnitID`

## Affected Files

- `backend/cmd/server/bootstrap/01-default-resources.sh` (line ~74)
- `backend/cmd/server/bootstrap/02-sample-resources.sh` (line ~58)

## Solution

Use the `/organization-units/tree/{handle}` API endpoint to fetch OUs by their specific handle instead of listing all OUs and picking the first one:

**Before:**
```bash
RESPONSE=$(thunder_api_call GET "/organization-units")
DEFAULT_OU_ID=$(echo "$BODY" | grep -o '"id":"[^"]*"' | head -1 | cut -d'"' -f4)
```

**After:**
```bash
RESPONSE=$(thunder_api_call GET "/organization-units/tree/default")
DEFAULT_OU_ID=$(echo "$BODY" | grep -o '"id":"[^"]*"' | head -1 | cut -d'"' -f4)
```

This ensures:
- ✅ Idempotent script execution
- ✅ Correct OU retrieval by handle
- ✅ No dependency on response ordering

## Steps to Reproduce

1. Run bootstrap scripts first time → Success
2. Run bootstrap scripts second time → Failure with "Organization unit mismatch" error

## Impact

- Prevents re-running bootstrap scripts during development/testing
- Breaks multi-environment setups where scripts need to be idempotent
- Affects any scenario where OUs are created in different orders

## Priority

Medium - Affects developer experience and testing workflows

--- a/backend/cmd/server/bootstrap/01-default-resources.ps1
+++ b/backend/cmd/server/bootstrap/01-default-resources.ps1
@@ -76,12 +76,12 @@ if ($response.StatusCode -eq 201 -or $response.StatusCode -eq 200) {
 }
 elseif ($response.StatusCode -eq 409) {
     Log-Warning "Organization unit already exists, retrieving OU ID..."
-    # Get existing OU ID
-    $response = Invoke-ThunderApi -Method GET -Endpoint "/organization-units"
+    # Get existing OU ID by handle to ensure we get the correct "default" OU
+    $response = Invoke-ThunderApi -Method GET -Endpoint "/organization-units/tree/default"
 
     if ($response.StatusCode -eq 200) {
         $body = $response.Body | ConvertFrom-Json
-        $DEFAULT_OU_ID = $body.organizationUnits[0].id
+        $DEFAULT_OU_ID = $body.id
         if ($DEFAULT_OU_ID) {
             Log-Success "Found OU ID: $DEFAULT_OU_ID"
         }
@@ -91,7 +91,7 @@ elseif ($response.StatusCode -eq 409) {
         }
     }
     else {
-        Log-Error "Failed to fetch organization units (HTTP $($response.StatusCode))"
+        Log-Error "Failed to fetch organization unit by handle 'default' (HTTP $($response.StatusCode))"
         exit 1
     }
 }

--- a/backend/cmd/server/bootstrap/01-default-resources.sh
+++ b/backend/cmd/server/bootstrap/01-default-resources.sh
@@ -69,8 +69,8 @@ if [[ "$HTTP_CODE" == "201" ]] || [[ "$HTTP_CODE" == "200" ]]; then
     fi
 elif [[ "$HTTP_CODE" == "409" ]]; then
     log_warning "Organization unit already exists, retrieving OU ID..."
-    # Get existing OU ID
-    RESPONSE=$(thunder_api_call GET "/organization-units")
+    # Get existing OU ID by handle to ensure we get the correct "default" OU
+    RESPONSE=$(thunder_api_call GET "/organization-units/tree/default")
     HTTP_CODE="${RESPONSE: -3}"
     BODY="${RESPONSE%???}"
 
@@ -83,7 +83,7 @@ elif [[ "$HTTP_CODE" == "409" ]]; then
             exit 1
         fi
     else
-        log_error "Failed to fetch organization units (HTTP $HTTP_CODE)"
+        log_error "Failed to fetch organization unit by handle 'default' (HTTP $HTTP_CODE)"
         exit 1
     fi
 else

--- a/backend/cmd/server/bootstrap/02-sample-resources.ps1
+++ b/backend/cmd/server/bootstrap/02-sample-resources.ps1
@@ -67,16 +67,14 @@ if ($response.StatusCode -eq 201 -or $response.StatusCode -eq 200) {
 }
 elseif ($response.StatusCode -eq 409) {
     Log-Warning "Customers organization unit already exists, retrieving ID..."
-    $response = Invoke-ThunderApi -Method GET -Endpoint "/organization-units"
+    # Get existing OU ID by handle to ensure we get the correct "customers" OU
+    $response = Invoke-ThunderApi -Method GET -Endpoint "/organization-units/tree/$customerOuHandle"
     if ($response.StatusCode -eq 200) {
         $body = $response.Body | ConvertFrom-Json
-        $customersOu = $body.organizationUnits | Where-Object { $_.handle -eq $customerOuHandle } | Select-Object -First 1
-        if ($customersOu) {
-            $CUSTOMER_OU_ID = $customersOu.id
-        }
+        $CUSTOMER_OU_ID = $body.id
     }
     else {
-        Log-Error "Failed to fetch organization units (HTTP $($response.StatusCode))"
+        Log-Error "Failed to fetch organization unit by handle '$customerOuHandle' (HTTP $($response.StatusCode))"
         Write-Host "Response: $($response.Body)"
         exit 1
     }

--- a/backend/cmd/server/bootstrap/02-sample-resources.sh
+++ b/backend/cmd/server/bootstrap/02-sample-resources.sh
@@ -54,29 +54,15 @@ if [[ "$HTTP_CODE" == "201" ]] || [[ "$HTTP_CODE" == "200" ]]; then
     CUSTOMER_OU_ID=$(echo "$BODY" | grep -o '"id":"[^"]*"' | head -1 | cut -d'"' -f4)
 elif [[ "$HTTP_CODE" == "409" ]]; then
     log_warning "Customers organization unit already exists, retrieving ID..."
-    RESPONSE=$(thunder_api_call GET "/organization-units")
+    # Get existing OU ID by handle to ensure we get the correct "customers" OU
+    RESPONSE=$(thunder_api_call GET "/organization-units/tree/${CUSTOMER_OU_HANDLE}")
     HTTP_CODE="${RESPONSE: -3}"
     BODY="${RESPONSE%???}"
 
     if [[ "$HTTP_CODE" == "200" ]]; then
-        CUSTOMER_OU_ID=$(
-            CUSTOMER_OU_HANDLE_VALUE="$CUSTOMER_OU_HANDLE" \
-            BODY_JSON="$BODY" \
-            python3 - <<'PY'
-import json
-import os
-
-body = json.loads(os.environ["BODY_JSON"])
-handle = os.environ["CUSTOMER_OU_HANDLE_VALUE"]
-
-for ou in body.get("organizationUnits", []):
-    if ou.get("handle") == handle:
-        print(ou.get("id", ""), end="")
-        break
-PY
-        )
+        CUSTOMER_OU_ID=$(echo "$BODY" | grep -o '"id":"[^"]*"' | head -1 | cut -d'"' -f4)
     else
-        log_error "Failed to fetch organization units (HTTP $HTTP_CODE)"
+        log_error "Failed to fetch organization unit by handle '${CUSTOMER_OU_HANDLE}' (HTTP $HTTP_CODE)"
         echo "Response: $BODY"
         exit 1
     fi

